schema/resolvers/tools.js

```diff
@@ -34,9 +27,40 @@ const toolsResolver = {
         },
       };
     },
-    tool: async (_, args, { toolLoader }, info) => {
-      console.log('ran tool');
-      const tool = await mongoDao.getOneDoc(database, 'tools', '_id', ObjectID(args._id));
+    getToolsByCategory: async (parent, { cursor, limit = 9, category }, context, info) => {
+      const cursorOptions = cursor
+        ? [
+            {
+              createdAt: {
+                $lt: fromCursorHash(cursor),
+              },
+            },
+          ]
+        : {};
+
+      const pipeline = [
+        {
+          $match: {
+            $and: [{ category: { $in: [category] } }],
+          },
+        },
+        { $sort: { createdAt: -1 } },
+        { $limit: limit + 1 },
+      ];
+
+      const tools = await mongoDao.pool.db(database).collection('tools').aggregate(pipeline, cursorOptions).toArray();
+      const hasNextPage = tools.length > limit;
```

Comment:
what happen if database is down or something else happened that make tools on line 51 undefined? then it would break. does it have a validator and also error handler?